cli/src/cmd/forge/script/mod.rs

```diff
@@ -699,6 +711,36 @@ impl ScriptConfig {
     fn target_contract(&self) -> &ArtifactId {
         self.target_contract.as_ref().expect("should exist after building")
     }
+
+    /// Checks if the RPCs used point to chains that support EIP-3855.
+    /// If not, warns the user.
+    async fn check_shanghai_support(&self) -> eyre::Result<()> {
+        let chain_ids = self.total_rpcs.iter().map(|rpc| async move {
+            if let Ok(provider) = ethers::providers::Provider::<Http>::try_from(rpc) {
+                match provider.get_chainid().await {
+                    Ok(chain_id) => match TryInto::<Chain>::try_into(chain_id) {
+                        Ok(chain) => return SHANGHAI_ENABLED_CHAINS.contains(&chain),
+                        Err(_) => return false,
+                    },
+                    Err(_) => return false,
+                }
+            }
+            false
+        });
+
+        let chain_ids_supported =
+            future::join_all(chain_ids).await.into_iter().any(|result| result);
+
+        // At least one chain ID unsupported.
+        if !chain_ids_supported {
+            let msg = "\
+EIP-3855 is not supported in one or more of the RPCs used.
+Contracts deployed with a Solidity version equal or higher than 0.8.20 might not work properly.
+For more information, please see https://eips.ethereum.org/EIPS/eip-3855";
+            shell::println(Paint::yellow(msg))?;
```

Comment:
it'd be good to include the chain id that does not support it her, otherwise the user needs to identify this manually.

Reply:
sweet, here's how it would look:
<img width="700" alt="Screen Shot 2023-05-04 at 14 09 58" src="https://user-images.githubusercontent.com/26014927/236291781-e464ac97-b532-4b4f-aa87-8a3e5108e28d.png">

Reply:
Ah will quickly change the message so it points out these are chain IDs.